### ISSUE TITLE
Encryption and decryption errors occur repeatedly

### ISSUE THREAD

**Issue by yuanchangjian** [kjur/jsrsasign#582]
`
export class LYRSA extends LYObject implements ILYRSA {
	private _publicKey: string;

	protected async getPublicKey(): Promise<string> {
		if (!this._publicKey) {
			const publicKey = await rootHttp.getPublicKey();
			this._publicKey = publicKey.pubSubjectKey;
		}
		return this._publicKey;
	}

	protected async getPrivateKey(): Promise<string> {
		throw new Error('Method not implemented.');
	}

	async encrypt(data: string): Promise<string> {
		const publicKey = await this.getPublicKey();
		const key = '-----BEGIN PUBLIC KEY-----\n' + publicKey + '\n-----END PUBLIC KEY-----';
		return hextob64(KJUR.crypto.Cipher.encrypt(data, KEYUTIL.getKey(key) as RSAKey, 'RSAOAEP256'));
	}

	async decrypt(data: string): Promise<string> {
		const privateKey = await this.getPrivateKey();
		const key = "-----BEGIN RSA PRIVATE KEY-----\n"+ privateKey + "\n" + "-----END RSA PRIVATE KEY-----"
		return KJUR.crypto.Cipher.decrypt(b64utohex(data), KEYUTIL.getKey(key) as RSAKey, 'RSAOAEP256');
	}
}
`

`
const key = "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA6kw0JFxZqgxISVe54G+ZHeyOoIpGr++DqplOQuijlR48GhkPjYd99J5DwQQXwb8lbNUPU7iwlQ7sqfNAMjk/CXmMM56th/jPlMEydZuzn+/SmNMPsVUd+Cn3dUMQMhZZZcNkFZuVuIzTSQsrjUah7dGdiozsKyriBbksiehO2i7Vc61PlTJOU/1PCgKHjsaw5qzB9Qfh2R/esM9miuiOKEiA4kupij3w7GTwi6NZ/VjSD+2uDgCpaIWjhxJQIT6kzE6hjx12wV94eMkKvkv9F/sZGmCBefFVzvahUmINTe4xGdainQ7aHZkYdHWJpnFm7N/u/MdNMAmawLv8RbKlwQIDAQAB"
const pKey = "MIIEpAIBAAKCAQEA6kw0JFxZqgxISVe54G+ZHeyOoIpGr++DqplOQuijlR48GhkPjYd99J5DwQQXwb8lbNUPU7iwlQ7sqfNAMjk/CXmMM56th/jPlMEydZuzn+/SmNMPsVUd+Cn3dUMQMhZZZcNkFZuVuIzTSQsrjUah7dGdiozsKyriBbksiehO2i7Vc61PlTJOU/1PCgKHjsaw5qzB9Qfh2R/esM9miuiOKEiA4kupij3w7GTwi6NZ/VjSD+2uDgCpaIWjhxJQIT6kzE6hjx12wV94eMkKvkv9F/sZGmCBefFVzvahUmINTe4xGdainQ7aHZkYdHWJpnFm7N/u/MdNMAmawLv8RbKlwQIDAQABAoIBAHHckKZu/uAl0edaq6lHcqjZMkMQj+OmSV1Z3W+ksQKhy+F+9lD4XJzJYPHnNEhixSRLsoQkSCRYOWkqHrOJoHnRSPI5FenzX6XR8dy3yyZZnQXJ5GFwlPzqICkray58L/vFZdkeYDdssK1ZsVOZ1M3HRYF0lL0xvN+9KsjmTZO3qtG6yt8pL7EZW19AfZFEuNdD6EEV2cEB4zPT0gRWh+uAIIu+XYnSy4dEAqksQM/iiJ+CMrxJceTatfKLPXUdC8qI+IuUiUDwZLsu15C3G0E06DMk10wCSZjZEgDTM5hqWPUWahpUh5lQDW64zlZM3qm/cxcqfA3+VOMlagc/I7ECgYEA+Ft1QoiXPoCRCVYyJ8eiBS7xuMLk91kKyLF8vzi4XwXmy46T2TuPMGRDw+gGl65vS2TfHaD23VS12G90Jv6YyBOEo/A4g9ImpSwblI2kwl3vnR7t5R46gWUveed54QQec9UloEzfB9s0pXDhP9Eng1yo4yOFhC/cQKDd0I9jNUUCgYEA8YH8Mhq+dDmwaf8tvoG/3cEAF/g45qrY1wfPjFYhZGrgGKkpc6DmYTeugYCJFx+VBOtrA7O23lAjuAxhGLYE74KE0tzNQzSXllG0aJG++1Ive6DNu7UXXELaWYjxacojMbPH2e7HMTJTzgVvIOtTtBo5sSr3R6JpX0n+AfEXIE0CgYAR12AhG3biApJovUx1Fx5IiQrMXxPbwQXJTRbNFeII0X+299MfM+whadKEbFKs6tcL738nvbx+QeueVMNfWcLX/K72p/cnYR1tmDGTBmnjnKPuAZ7NatKBISe79ikHRSh64cEGNv9+rPynNei4kOnpwepQBuk8icOk0OmeZe6WkQKBgQDO221uZSXYW0lVoPJHiM8UW6l+3meue4u7sr79DxEO99P2i34rwhdSEeVwmu3RVoXHbvV8eEQb6zyM+l14Uv+rPnoSOnm+JJ/qaxVBtASHM3/kPnm5W4Uky0ZHBlrxJ+g5PCnj6vfGlbrebh0fbzGExCnCLHEddGkY60Gb3CrYyQKBgQCqAqg9InHjP0ma+Co7ptHYAzm46+Jq7QowsCl07SoTujVf9qaYpohdhqp8kXVonwUrQQ+PEwvtK4ORGA3FIPgfDXaV5g7gThsTGTZ11GAKuzdtQSCPRuRR1+wcxZaGF0WfLJw1Jvd8vKEt2qBJ+Jrlj4rnH2mZ9n3ZFEof14L7iQ=="

@register('LYTestRSA')
class LYTestRSA extends LYRSA {
	protected getPublicKey(): Promise<string> {
		return Promise.resolve(key);
	}

	protected getPrivateKey(): Promise<string> {
		return Promise.resolve(pKey);
	}
}

describe.only('LYEnv', function () {
	this.timeout(3000000);

	const RSA = new LYTestRSA();

	it('encrypt & decrypt', async () => {
		for (let i = 0; i < 10000; i++) {
			console.log(i)
			const data = 'ngzCT7HX5GK3gLaT';
			const encrypt = await RSA.encrypt(data);
			const decrypt = await RSA.decrypt(encrypt);
			assert.equal(decrypt, data);
		}
	});
});
`

![image](https://user-images.githubusercontent.com/14868050/231037838-21723021-4c7e-4582-9e7b-cc4864a0934e.png)

**Comment by yuanchangjian** [comment]
import { KJUR, hextob64, KEYUTIL, b64utohex, RSAKey } from 'jsrsasign';

**Comment by yuanchangjian** [comment]
![image](https://user-images.githubusercontent.com/14868050/231039697-408c6aeb-4094-48d8-8f3b-88acb81a8197.png)

**Comment by yuanchangjian** [comment]
![image](https://user-images.githubusercontent.com/14868050/231040783-f131bd70-6f27-43cb-b46b-70edae88f0a7.png)

**Comment by kjur** [comment]
I've tried encryption and decryption with your key and data without your Promise codes 100 times. Then they didn't fail.

```JavaScript
#!/usr/bin/env node

var rs = require("jsrsasign");

const key = "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA6kw0JFxZqgxISVe54G+ZHeyOoIpGr++DqplOQuijlR48GhkPjYd99J5DwQQXwb8lbNUPU7iwlQ7sqfNAMjk/CXmMM56th/jPlMEydZuzn+/SmNMPsVUd+Cn3dUMQMhZZZcNkFZuVuIzTSQsrjUah7dGdiozsKyriBbksiehO2i7Vc61PlTJOU/1PCgKHjsaw5qzB9Qfh2R/esM9miuiOKEiA4kupij3w7GTwi6NZ/VjSD+2uDgCpaIWjhxJQIT6kzE6hjx12wV94eMkKvkv9F/sZGmCBefFVzvahUmINTe4xGdainQ7aHZkYdHWJpnFm7N/u/MdNMAmawLv8RbKlwQIDAQAB";

const pKey = "MIIEpAIBAAKCAQEA6kw0JFxZqgxISVe54G+ZHeyOoIpGr++DqplOQuijlR48GhkPjYd99J5DwQQXwb8lbNUPU7iwlQ7sqfNAMjk/CXmMM56th/jPlMEydZuzn+/SmNMPsVUd+Cn3dUMQMhZZZcNkFZuVuIzTSQsrjUah7dGdiozsKyriBbksiehO2i7Vc61PlTJOU/1PCgKHjsaw5qzB9Qfh2R/esM9miuiOKEiA4kupij3w7GTwi6NZ/VjSD+2uDgCpaIWjhxJQIT6kzE6hjx12wV94eMkKvkv9F/sZGmCBefFVzvahUmINTe4xGdainQ7aHZkYdHWJpnFm7N/u/MdNMAmawLv8RbKlwQIDAQABAoIBAHHckKZu/uAl0edaq6lHcqjZMkMQj+OmSV1Z3W+ksQKhy+F+9lD4XJzJYPHnNEhixSRLsoQkSCRYOWkqHrOJoHnRSPI5FenzX6XR8dy3yyZZnQXJ5GFwlPzqICkray58L/vFZdkeYDdssK1ZsVOZ1M3HRYF0lL0xvN+9KsjmTZO3qtG6yt8pL7EZW19AfZFEuNdD6EEV2cEB4zPT0gRWh+uAIIu+XYnSy4dEAqksQM/iiJ+CMrxJceTatfKLPXUdC8qI+IuUiUDwZLsu15C3G0E06DMk10wCSZjZEgDTM5hqWPUWahpUh5lQDW64zlZM3qm/cxcqfA3+VOMlagc/I7ECgYEA+Ft1QoiXPoCRCVYyJ8eiBS7xuMLk91kKyLF8vzi4XwXmy46T2TuPMGRDw+gGl65vS2TfHaD23VS12G90Jv6YyBOEo/A4g9ImpSwblI2kwl3vnR7t5R46gWUveed54QQec9UloEzfB9s0pXDhP9Eng1yo4yOFhC/cQKDd0I9jNUUCgYEA8YH8Mhq+dDmwaf8tvoG/3cEAF/g45qrY1wfPjFYhZGrgGKkpc6DmYTeugYCJFx+VBOtrA7O23lAjuAxhGLYE74KE0tzNQzSXllG0aJG++1Ive6DNu7UXXELaWYjxacojMbPH2e7HMTJTzgVvIOtTtBo5sSr3R6JpX0n+AfEXIE0CgYAR12AhG3biApJovUx1Fx5IiQrMXxPbwQXJTRbNFeII0X+299MfM+whadKEbFKs6tcL738nvbx+QeueVMNfWcLX/K72p/cnYR1tmDGTBmnjnKPuAZ7NatKBISe79ikHRSh64cEGNv9+rPynNei4kOnpwepQBuk8icOk0OmeZe6WkQKBgQDO221uZSXYW0lVoPJHiM8UW6l+3meue4u7sr79DxEO99P2i34rwhdSEeVwmu3RVoXHbvV8eEQb6zyM+l14Uv+rPnoSOnm+JJ/qaxVBtASHM3/kPnm5W4Uky0ZHBlrxJ+g5PCnj6vfGlbrebh0fbzGExCnCLHEddGkY60Gb3CrYyQKBgQCqAqg9InHjP0ma+Co7ptHYAzm46+Jq7QowsCl07SoTujVf9qaYpohdhqp8kXVonwUrQQ+PEwvtK4ORGA3FIPgfDXaV5g7gThsTGTZ11GAKuzdtQSCPRuRR1+wcxZaGF0WfLJw1Jvd8vKEt2qBJ+Jrlj4rnH2mZ9n3ZFEof14L7iQ==";

var data = "ngzCT7HX5GK3gLaT";

//console.log(rs.VERSION);

let pubkey = rs.KEYUTIL.getKey(rs.b64topem(key, "PUBLIC KEY"));
let prvkey = rs.KEYUTIL.getKey(rs.b64topem(pKey, "RSA PRIVATE KEY"));

let countSuccess = 0;
for (let i = 0; i < 100; i++) {
    //console.log(data);
    let hEnc = rs.KJUR.crypto.Cipher.encrypt(data, pubkey, "RSAOAEP256");
    //console.log(hEnc);
    let sDec = rs.KJUR.crypto.Cipher.decrypt(hEnc, prvkey, "RSAOAEP256");
    //console.log(sDec);
    if (data == sDec) countSuccess++;
}
console.log(`SUCCESS=${countSuccess} FAIL=${100 - countSuccess}`);
```

This shows:
```
SUCCESS=100 FAIL=0
```

It seems to having issue in your code. Thank you.

**Comment by yuanchangjian** [comment]
@kjur

**Comment by kjur** [comment]
Hello @davedoesdev ,

Thank you for your contribution for RSA OAEP encryption and decryption. As reported above, your OAEP encryption function will occasionally generates wrong encrypted data which can't be decrypted. Failure rate is approxmately 0.5% or so.

Could you help me to fix it?

Here is a test code to reproduce the issue:
```JavaScript
#!/usr/bin/env node

var rs = require("jsrsasign");

const key = "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA6kw0JFxZqgxISVe54G+ZHeyOoIpGr++DqplOQuijlR48GhkPjYd99J5DwQQXwb8lbNUPU7iwlQ7sqfNAMjk/CXmMM56th/jPlMEydZuzn+/SmNMPsVUd+Cn3dUMQMhZZZcNkFZuVuIzTSQsrjUah7dGdiozsKyriBbksiehO2i7Vc61PlTJOU/1PCgKHjsaw5qzB9Qfh2R/esM9miuiOKEiA4kupij3w7GTwi6NZ/VjSD+2uDgCpaIWjhxJQIT6kzE6hjx12wV94eMkKvkv9F/sZGmCBefFVzvahUmINTe4xGdainQ7aHZkYdHWJpnFm7N/u/MdNMAmawLv8RbKlwQIDAQAB";

const pKey = "MIIEpAIBAAKCAQEA6kw0JFxZqgxISVe54G+ZHeyOoIpGr++DqplOQuijlR48GhkPjYd99J5DwQQXwb8lbNUPU7iwlQ7sqfNAMjk/CXmMM56th/jPlMEydZuzn+/SmNMPsVUd+Cn3dUMQMhZZZcNkFZuVuIzTSQsrjUah7dGdiozsKyriBbksiehO2i7Vc61PlTJOU/1PCgKHjsaw5qzB9Qfh2R/esM9miuiOKEiA4kupij3w7GTwi6NZ/VjSD+2uDgCpaIWjhxJQIT6kzE6hjx12wV94eMkKvkv9F/sZGmCBefFVzvahUmINTe4xGdainQ7aHZkYdHWJpnFm7N/u/MdNMAmawLv8RbKlwQIDAQABAoIBAHHckKZu/uAl0edaq6lHcqjZMkMQj+OmSV1Z3W+ksQKhy+F+9lD4XJzJYPHnNEhixSRLsoQkSCRYOWkqHrOJoHnRSPI5FenzX6XR8dy3yyZZnQXJ5GFwlPzqICkray58L/vFZdkeYDdssK1ZsVOZ1M3HRYF0lL0xvN+9KsjmTZO3qtG6yt8pL7EZW19AfZFEuNdD6EEV2cEB4zPT0gRWh+uAIIu+XYnSy4dEAqksQM/iiJ+CMrxJceTatfKLPXUdC8qI+IuUiUDwZLsu15C3G0E06DMk10wCSZjZEgDTM5hqWPUWahpUh5lQDW64zlZM3qm/cxcqfA3+VOMlagc/I7ECgYEA+Ft1QoiXPoCRCVYyJ8eiBS7xuMLk91kKyLF8vzi4XwXmy46T2TuPMGRDw+gGl65vS2TfHaD23VS12G90Jv6YyBOEo/A4g9ImpSwblI2kwl3vnR7t5R46gWUveed54QQec9UloEzfB9s0pXDhP9Eng1yo4yOFhC/cQKDd0I9jNUUCgYEA8YH8Mhq+dDmwaf8tvoG/3cEAF/g45qrY1wfPjFYhZGrgGKkpc6DmYTeugYCJFx+VBOtrA7O23lAjuAxhGLYE74KE0tzNQzSXllG0aJG++1Ive6DNu7UXXELaWYjxacojMbPH2e7HMTJTzgVvIOtTtBo5sSr3R6JpX0n+AfEXIE0CgYAR12AhG3biApJovUx1Fx5IiQrMXxPbwQXJTRbNFeII0X+299MfM+whadKEbFKs6tcL738nvbx+QeueVMNfWcLX/K72p/cnYR1tmDGTBmnjnKPuAZ7NatKBISe79ikHRSh64cEGNv9+rPynNei4kOnpwepQBuk8icOk0OmeZe6WkQKBgQDO221uZSXYW0lVoPJHiM8UW6l+3meue4u7sr79DxEO99P2i34rwhdSEeVwmu3RVoXHbvV8eEQb6zyM+l14Uv+rPnoSOnm+JJ/qaxVBtASHM3/kPnm5W4Uky0ZHBlrxJ+g5PCnj6vfGlbrebh0fbzGExCnCLHEddGkY60Gb3CrYyQKBgQCqAqg9InHjP0ma+Co7ptHYAzm46+Jq7QowsCl07SoTujVf9qaYpohdhqp8kXVonwUrQQ+PEwvtK4ORGA3FIPgfDXaV5g7gThsTGTZ11GAKuzdtQSCPRuRR1+wcxZaGF0WfLJw1Jvd8vKEt2qBJ+Jrlj4rnH2mZ9n3ZFEof14L7iQ==";

var data = "ngzCT7HX5GK3gLaT";

let pubkey = rs.KEYUTIL.getKey(rs.b64topem(key, "PUBLIC KEY"));
let prvkey = rs.KEYUTIL.getKey(rs.b64topem(pKey, "RSA PRIVATE KEY"));

let numTry = 1000;
let countSuccess = 0;
let countException = 0;
for (let i = 0; i < numTry; i++) {
    let hEnc, sDec;
    hEnc = pubkey.encryptOAEP(data, "sha256");
    try {
	sDec = prvkey.decrypt(hEnc, "sha256");
	if (data == sDec) countSuccess++;
    } catch(ex) {
	console.log("DECRYPT FAILED");
	console.log(hEnc);
	countException++;
    }
}
console.log(`SUCCESS=${countSuccess} FAIL=${numTry - countSuccess - countException} EXCEPTION=${countException}`);
```

And below is its output:
```
DECRYPT FAILED
8a5dc6126b116e37e721332097399ece64203841e534a196d6cd2fdc476850639edcf841c412f0da89947801b0efa7daf798d66e1ac552621263e958dd7abe65c61c4dcfc774faab741129b4a9bbea5f22b6f91b5c02301c3c7ac79a0ee713c5333d1d6e287e81f3a23cfec75bd1a95538ce54ffee9bef5fdc5074105747cebf7b7778a12f3b6622c54cd599e51ecd5680ae76356f8b9ee10841061f2484c44dbddb2252433022ff55384080ffa5ceb1dfe7942ac9421c1007443e738b2d6251b3098ec34e73de04eb2ea41c66f18ed6921187354ac67baa5278b1e0033cc423dc35a47b29fc9c4a627f830c6287832212ad4c5466c00e1750d12698e377ac
DECRYPT FAILED
93ca7ab7216f82ca599c49e80c2e5178883479b1a0261f32cc68fd62ceec03c2eacd8ae02165c9f740d4825d056c06e344c8587d936a954098d9bb0752328517fb18d63a1ec12c6845656a9cc12aacfcf4702cf91a2ccc6e992368084eb1fade413b6fa08eddcab36919ed89ba1334365db2b8c3fd7d953d04e579b3b3877193cbe9f4f8bd64312c90d0433dbd5e983967692fedd05ece7958cd3c9043d15b9d8dac64b9f21b10a0f1e159a3aaf4bd6d2431151655997d9cbaf0c30d0779eab87c1a6af99a1f8f4f1d26ebe01996c849208348ff0269de0b5902defdf3b9dddc69eb5a4985e49fba00b4aaa21aec0dfce12988dafdd6d54e9c2c17cad07fd7
DECRYPT FAILED
635ab3087da0e985fb3e3e689b8f7148975969b0991342177395454acb54df4cd6ff1b69f3c8c520123bcd505bd0ba1e6f8f08c840554c1209a5688d33179ef4c0f38047193ec9c1bd819b25bcdbd3e043124e097d846e4aacbc7fa7cce69d636d0c096a5f6141cbd66adf0547c9903f53b3e81e475c8cc15f3f7fb5be6cea21f22cc779a1694a51b364b4834be37370dc29da4cd3b5a3d7f1e6341915e5c72ef5b54042b5da48a19604ec5b0ae379b4e663efa6ba66b3cfbb10c5a30da67a0c4a0fa694c7e34849749c3490d8a8e305d5a683c2c41bff4945d5ef8f83ae7ce700988cac02a9df19fc0c2893c4066f896b83040a7df7e04b1f029a92916e48
DECRYPT FAILED
db02b8ac65d878d47cb401b28ce01a430436654ef982a1b137338516f8f2f963f173d3b38b8bbdaceb50a6b064a683a32220f2541a11ea2bc4d81c6bc9b664e3765dc92a9242befb07373f4a9fb099fc11aa45a67847f143397e629ca6e0c0272b504d68a4e5609c463d76e11695984eee077e1bf35fe1bad8e036eb1ffa4c24c5e6c69bed4bf6bc67f942f1065272fde16eb239a297120c975632b6a3b728d1ffceb61d283712d5257147bcd05b7cef198bcdd7997bc0b153bbf109fa52d89b57f2dc19c687f70351917b9e2b906580bfd3ed9ab5f4dd43f644613b3ccd5ef0e2876345a7ab56092a1496786cdc7e64dd67703ec60d19e1f01e2f7fc065e2
DECRYPT FAILED
fad109f381c32846ab1ca2bda44c9e0f95566893b86af1cca5c27143e6e3d3bdf0c7f9e2c7013fb6c353e280bf61c7053f4fa434527d62ab3f5cb873603b81f322244c864ac62614b5351403064d120e5644e51089e561ded1092490a11dfe5aa1f826b6780a2a403cd3dfebd2c5f3671adc3de232fd155705c8a7820f3dfeebfab8b83e5eab5482da08e7a5684a8e84f3af320335ebb158f441f9303815f34e7f2990f3e3c4c26fa5848aa04cc733d00916b62899b4f4c7551310c0b33376c9a0fd5e65d631ce3540ad96f8d9bf113a25c48ecc22e60582be6d60b16728f34ddc3afe9d042a854a0a6091d54d563c083eb6280de4b61fc52a95391a117024
DECRYPT FAILED
e9cea971eea751d41d3faff5430e55a2001529b126833580a56af6151d17ac693be9010fb9495702ac61357392c9631cd42f8f36a60557e88a1cfe1932d26fcd3eab96411218160fce1f1a189aa70649298ec70c25dbcf81689bf3ef0fcbe0be50c2742eb85801b859111a5d86e5082725b0289ca104b3e0a8e084c93b57eaef6d674ea5146ee61342dd08e9224236e82ed4cfb06a16ddd42e960edebff5d8387731da0a6bed20ac3ecedf0074acde380c18f12081919e013e93fd6b062dc3a42e13899daa47bbd3995ff6ba127536518707ae0e624e4882d545aae4abf06310871017beb9318b4f57862c1ca99bb171a863f7b237d84a2dc04fd44c6ee0cf
DECRYPT FAILED
b9dcd01d2367d60ee5cd1da9cd38ffded4eaccdf108b8aeddfa9c8092fe2186a73c9e3f0df0c8c0e1e2d0170a58ee9a55b7b3780db58097fc7aaef8e63f9687c0d878c80555509a824ddd2aa26f8db432aecbe35e330c6be9f79b6c46826b2843e40e7b3d61462c23adc188e0abf3f37a7a32cced4c4eca37d4279091f1aa0b081cc0563fe2a6dd6c8616adeaf29be33dcfd738682e1b7c7e761df79f2542e410127ec53d2e0dbfdca971fbfdf3e01bc82ceb1cafe6adaf7d9bccbc616f5126f4ce8cbe3b8b7cf3f53d16e6ac47b255830cbe0081b05f44dac16f66ea645ff8634a6abf123e9f3b1f4d56dfbdcb9c56993d8f63a57e83e955ba07ecd7b9630
SUCCESS=0 FAIL=993 EXCEPTION=7
```

**Comment by davedoesdev** [comment]
Sure, I'll take a look. Might be a few days though.

**Comment by davedoesdev** [comment]
How do I build `jsrsasign-all-min.js`?

**Comment by kjur** [comment]
Hi @davedoesdev , 

I've attached some script to generate jsrsasign.js or jsrsasign-all-min.js without minification.
[mktestlib.zip](https://github.com/kjur/jsrsasign/files/11229877/mktestlib.zip)

These scripts will generate "lib/jsrsasign.js" and "lib/jsrsasign-all-min.js".

USAGE

1. copy this archive to npm directory
   % cp mktestlib.zip npm/

2. unzip it
   % unzip mktestlib.zip

3. to generate lib/jsrsasign-all-min.js
   % mktestlib_libjsrsasignallmin.sh
   Then file "testlib_libjsrsasignallmin.js" will be generated.
   This can be replacement of "lib/jsrsasign-all-min.js".

4. to generate lib/jsrsasign.js
   % mktestlib_libjsrsasign.sh
   Then file "testlib_libjsrsasign.js" will be generated.
   This can be replacement of "lib/jsrsasign.js".

**Comment by kjur** [comment]
Hi @davedoesdev san,

Current lib/jsrsasign.js and lib/jsrsasign-all-min.js generator depend on old and wrong "YUI compressor" and I think it would be better not to introduce you such generator. For your debugging, above scripts may enough.

**Comment by kjur** [comment]
Otherwise I can write a QUnit test script for the issue like other test/qunit-*.html test scripts. Please ask me if you want it.

**Comment by kjur** [comment]
Hi @yuanchangjian , this issue was fixed in 10.8.2 release today.

Hi @davedoesdev san, thank you kindly for your help.

**Comment by kjur** [comment]
Just FYI. Here is mocha test code for 10,000 times RSA OAEP encryption and decryption. It takes too much time so that it is not included in the release:

``` JavaScript
// for issue 582, 583
var assert = require('assert');
var rs = require('../lib/jsrsasign.js');

const pubb64 = "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA6kw0JFxZqgxISVe54G+ZHeyOoIpGr++DqplOQuijlR48GhkPjYd99J5DwQQXwb8lbNUPU7iwlQ7sqfNAMjk/CXmMM56th/jPlMEydZuzn+/SmNMPsVUd+Cn3dUMQMhZZZcNkFZuVuIzTSQsrjUah7dGdiozsKyriBbksiehO2i7Vc61PlTJOU/1PCgKHjsaw5qzB9Qfh2R/esM9miuiOKEiA4kupij3w7GTwi6NZ/VjSD+2uDgCpaIWjhxJQIT6kzE6hjx12wV94eMkKvkv9F/sZGmCBefFVzvahUmINTe4xGdainQ7aHZkYdHWJpnFm7N/u/MdNMAmawLv8RbKlwQIDAQAB";
const prvb64 = "MIIEpAIBAAKCAQEA6kw0JFxZqgxISVe54G+ZHeyOoIpGr++DqplOQuijlR48GhkPjYd99J5DwQQXwb8lbNUPU7iwlQ7sqfNAMjk/CXmMM56th/jPlMEydZuzn+/SmNMPsVUd+Cn3dUMQMhZZZcNkFZuVuIzTSQsrjUah7dGdiozsKyriBbksiehO2i7Vc61PlTJOU/1PCgKHjsaw5qzB9Qfh2R/esM9miuiOKEiA4kupij3w7GTwi6NZ/VjSD+2uDgCpaIWjhxJQIT6kzE6hjx12wV94eMkKvkv9F/sZGmCBefFVzvahUmINTe4xGdainQ7aHZkYdHWJpnFm7N/u/MdNMAmawLv8RbKlwQIDAQABAoIBAHHckKZu/uAl0edaq6lHcqjZMkMQj+OmSV1Z3W+ksQKhy+F+9lD4XJzJYPHnNEhixSRLsoQkSCRYOWkqHrOJoHnRSPI5FenzX6XR8dy3yyZZnQXJ5GFwlPzqICkray58L/vFZdkeYDdssK1ZsVOZ1M3HRYF0lL0xvN+9KsjmTZO3qtG6yt8pL7EZW19AfZFEuNdD6EEV2cEB4zPT0gRWh+uAIIu+XYnSy4dEAqksQM/iiJ+CMrxJceTatfKLPXUdC8qI+IuUiUDwZLsu15C3G0E06DMk10wCSZjZEgDTM5hqWPUWahpUh5lQDW64zlZM3qm/cxcqfA3+VOMlagc/I7ECgYEA+Ft1QoiXPoCRCVYyJ8eiBS7xuMLk91kKyLF8vzi4XwXmy46T2TuPMGRDw+gGl65vS2TfHaD23VS12G90Jv6YyBOEo/A4g9ImpSwblI2kwl3vnR7t5R46gWUveed54QQec9UloEzfB9s0pXDhP9Eng1yo4yOFhC/cQKDd0I9jNUUCgYEA8YH8Mhq+dDmwaf8tvoG/3cEAF/g45qrY1wfPjFYhZGrgGKkpc6DmYTeugYCJFx+VBOtrA7O23lAjuAxhGLYE74KE0tzNQzSXllG0aJG++1Ive6DNu7UXXELaWYjxacojMbPH2e7HMTJTzgVvIOtTtBo5sSr3R6JpX0n+AfEXIE0CgYAR12AhG3biApJovUx1Fx5IiQrMXxPbwQXJTRbNFeII0X+299MfM+whadKEbFKs6tcL738nvbx+QeueVMNfWcLX/K72p/cnYR1tmDGTBmnjnKPuAZ7NatKBISe79ikHRSh64cEGNv9+rPynNei4kOnpwepQBuk8icOk0OmeZe6WkQKBgQDO221uZSXYW0lVoPJHiM8UW6l+3meue4u7sr79DxEO99P2i34rwhdSEeVwmu3RVoXHbvV8eEQb6zyM+l14Uv+rPnoSOnm+JJ/qaxVBtASHM3/kPnm5W4Uky0ZHBlrxJ+g5PCnj6vfGlbrebh0fbzGExCnCLHEddGkY60Gb3CrYyQKBgQCqAqg9InHjP0ma+Co7ptHYAzm46+Jq7QowsCl07SoTujVf9qaYpohdhqp8kXVonwUrQQ+PEwvtK4ORGA3FIPgfDXaV5g7gThsTGTZ11GAKuzdtQSCPRuRR1+wcxZaGF0WfLJw1Jvd8vKEt2qBJ+Jrlj4rnH2mZ9n3ZFEof14L7iQ==";
const data = "ngzCT7HX5GK3gLaT";

describe("rsa oaep enc dec loop", function() {
    describe("test", function() {
        let pubkey = rs.KEYUTIL.getKey(rs.b64topem(pubb64, "PUBLIC KEY"));
        let prvkey = rs.KEYUTIL.getKey(rs.b64topem(prvb64, "RSA PRIVATE KEY"));

        let numTry = 10000;
        //let numTry = 1;
        let countSuccess = 0;
        let countException = 0;
        for (let i = 0; i < numTry; i++) {
            let hEnc, sDec;
            hEnc = pubkey.encryptOAEP(data, "sha256");
            //console.log(hEnc);
            try {
                sDec = prvkey.decryptOAEP(hEnc, "sha256");
                if (data == sDec) countSuccess++;
            } catch(ex) {
                console.log(ex);
                countException++;
            }
        }
        //console.log(`${countSuccess} ${countException}`);
        assert.equal(countSuccess, numTry);
        assert.equal(countException, 0);
    });
});
```
